### DISCUSSION TITLE
Get all rows in collection based on user relation.

### DISCUSSION THREAD

**Discussion by nanasemanase** [pocketbase/pocketbase#3416] [Q&A] [answered]
I have two collections: one called 'users' and another called 'posts.' In the 'posts' collection, each row has a relation to the 'users' collection to identify the author. How can I retrieve all the posts made by a specific author without saving the posts they have made as relations in the 'users' column?

**Comment by ganigeorgiev** [comment, marked as answer]
If your "author" field in the "posts" collection is a _single_ relation, then you can directly compare its value against the user id you want to filter:

```js
const userId = "..."

const posts = await pb.collection("posts").getFullList({
    filter: `author="${userId}"`
})
```

**Reply by nanasemanase** [reply]
Wouldn't this become inefficient with a large amount of posts? Doesn't "filter" do this on the client.

**Reply by ganigeorgiev** [reply]
I'm not sure that I understand what do you mean. The `filter` expression is translated to a raw SQL. Internally the above would generate batched requests per 200 (_the batch size can be changed via the `batch` parameter_) with a SQL query like:

```sql
SELECT *
FROM posts
WHERE author = {:userId}
LIMIT 200
OFSET x
```

If you expect the author to have a lot of posts, you can fetch a paginated list with `getList()`:

```js
const userId = "..."

const paginatedPosts = await pb.collection("posts").getList(1, 30, {
    filter: `author="${userId}"`
})
```
_`getFullList` internally calls `getList` on batches until all items are retrieved._

**Reply by nanasemanase** [reply]
I see. I thought I read somewhere that it filtered the results on the users browser. Seems I was wrong. Thanks for clarifying 👍 😄
